Yosemite/Yosemite/Tools/Stats/LeaderboardStatsConverter.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+import class Aztec.HTMLParser
+import struct Aztec.Element
+
+/// Helper functions to  convert a `Leaderboard` type into a `TopEearnerStat`
+///
+struct LeaderboardStatsConverter {
+
+    /// Returns the ids of the products associated with a top products leaderboar
+    ///
+    static func topProductsIDs(from topProducts: Leaderboard) -> [Int64] {
+        return topProducts.rows.compactMap {
+            Self.infeerProductID(fromHTMLString: $0.subject.display)
+        }
+    }
+
+    /// Indicates what products are missing from a leaderboard in an array of stored products
+    ///
+    static func missingProductsIDs(from topProducts: Leaderboard, in storedProducts: [Product]) -> [Int64] {
```

Comment:
I thought about doing that, but passing an array of `Ints` semed less readable and more error-prone than passing the concrete object.

Since the products set is small I think It's fine to prefer readability over performance here :-)